go-controller/pkg/node/gateway_localnet.go

```diff
@@ -119,7 +119,13 @@ func (l *localPortWatcher) AddService(svc *kapi.Service) {
 }
 
 func (l *localPortWatcher) UpdateService(old, new *kapi.Service) {
-	if reflect.DeepEqual(new.Spec, old.Spec) {
+	if reflect.DeepEqual(new.Spec.Ports, old.Spec.Ports) &&
```

Comment:
LGTM....

But just a though. Could we move this check up to `gateway.AddService`?
Then we'd only need to do it once, and not have it done in multiple places?

Reply:
I thought about that, but I saw other handlers have different checks like look at port claim watcher:
```
	if reflect.DeepEqual(old.Spec.ExternalIPs, new.Spec.ExternalIPs) && reflect.DeepEqual(old.Spec.Ports, new.Spec.Ports) {
		return
	}
```